zephyr/lib/alloc.c

```diff
@@ -388,7 +388,8 @@ void *rballoc_align(uint32_t flags, uint32_t caps, size_t bytes,
 		heap = &l3_heap;
 		return (__sparse_force void *)l3_heap_alloc_aligned(heap, align, bytes);
 #else
-		k_panic();
+		tr_err(&zephyr_tr, "L3_HEAP not available.");
+		return NULL;
```

Comment:
will the propagate all the way up to the host?

Reply:
The default result of this call was to return NULL on allocation failure. I added k_panic() when L3_HEAP support was added to rballoc_allign(). Although this change exposed a bug in the IPC3 handling code, it seems that returning an allocation error is sufficient here.

Reply:
Sorry, not what I meant, I skipped the word "error". I mean't will the host properly be notified of this new allocation error path should it be hit.